docs/tutorials/tutorial2.md

```diff
@@ -165,6 +165,18 @@ value.
 
     `_addressLine1 = "address1"`
 
+When renaming the name of a field referring to a primary key (for example
+the `_addressForUser` field), remember to specify the field name as a
+`PrimaryKey` field like:
+
+`_addressForUser = UserId "user"`
+
+Given that in part one we defined `UserId` as a type alias:
+
+```haskell
+type UserId = PrimaryKey UserT Identity
```

Comment:
I think you're confusing the type synonym name and the constructor name here. They're not inherently related and this type synonym isn't what's used in the table definition (as it's not parametric over the `f` parameter).

Reply:
Hi, I see what you mean, but I was just quoting the definition as written in part one of the tutorial, assuming the reader is following the guide that is the type synonym you have up to that point. Are you suggesting we change the definition also in part one? something like `type UserPk f = PrimaryKey UserT f`?

Reply:
The relevant definition is the constructor name here:

```haskell
instance Table UserT where
   data PrimaryKey UserT f = UserId (Columnar f Text) deriving (Generic, Beamable)
   primaryKey = UserId . _userEmail
```

The fact that there is also a type synonym for `PrimaryKey UserT Identity` and that it shares the name `UserId` isn't why you need to use `UserId` to specify the name of the foreign key column.

Reply:
oooh I get it know